### ISSUE TITLE
false positive in block-no-redundant-nested-style-rules

### ISSUE THREAD

**Issue by Mister-Hope** [stylelint-scss/stylelint-scss#1190]
```scss
  [vp-content] {
    @include mixins.content-wrapper;

    // unexpected redundant nested style rule (block-no-redundant-nested-style-rules)Stylelint[block-no-redundant-nested-style-rules](https://stylelint.io/user-guide/rules/block-no-redundant-nested-style-rules)
    & {
      padding-top: 0;
    }
  }
```

This is a correct usage when mixins has at-selectors like `@media`, and this is a correct report for stylelint to alert for css files.

While, with stylelint-scss, issues are reported in scss files while stylelint-scss shall silence them.
